snapper/Snapshot.cc

```diff
@@ -794,7 +794,7 @@ namespace snapper
 
 	entries.erase(snapshot);
 
-	Hooks::delete_snapshot(snapper->subvolumeDir(), snapper->getFilesystem());
+	Hooks::delete_snapshot(snapper->subvolumeDir(), snapper->getFilesystem(), *snapshot);
```

Comment:
The erase above invalidates snapshot, thus the Hook must now be moved before the erase.